test/triangular.jl

```diff
@@ -650,6 +650,16 @@ end
             @test_throws "cannot set index in the upper triangular part" copyto!(A, B)
         end
     end
+
+    @testset "partly initialized unit triangular" begin
+        for T in (UnitUpperTriangular, UnitLowerTriangular)
+            isupper = T == UnitUpperTriangular
+            M = Matrix{BigFloat}(undef, 2, 2)
+            M[1+isupper,1+!isupper] = 3
+            U = T(M')
```

Comment:
What's the rationale here for the adjoint?

Reply:
This is to avoid the `StridedArray` method and hit the generic fallback. We may use the `Test.GenericArray` wrapper as well, to be doubly sure. I had used the adjoint from a failing test in another PR.